### DISCUSSION TITLE
"Cloudflare Workers are not Node" Confusion

### DISCUSSION THREAD

**Discussion by JGantts** [vikejs/vike#767] [Discussion] [answered]
Hello!

I'm trying to set up a SPA with SSR on Cloudflare Workers using vite-plugion-ssr. However I kept running into errors that pointed to the fact that apparently "Cloudflare Workers are not Node". Errors were about missing node modules and missing node itself.

I'm about to try to switch to Cloudflare Pages, but I was just hoping I could get some clarification as to why vite-plugin-ssr needs node despite not having it available on Cloudflare Workers?

Thanks for the code y'all wrote!
JGantts

**Comment by brillout** [comment, marked as answer]
See the first note of https://vite-plugin-ssr.com/cloudflare-workers and the issue it's referencing: https://github.com/brillout/vite-plugin-ssr/issues/445.
